core/domain/user_domain.py

```diff
@@ -326,6 +326,52 @@ def validate(self) -> None:
                 '%s is not a valid value for the dashboard display '
                 'preferences.' % (self.creator_dashboard_display_pref))
 
+        if not isinstance(self.subject_interests, list):
+            raise utils.ValidationError(
+                'Expected subject_interests to be a list.')
+
+        for interest in self.subject_interests:
+            if not isinstance(interest, str):
+                raise utils.ValidationError(
+                    'Expected each subject interest to be a string.')
+            if not interest:
+                raise utils.ValidationError(
+                    'Expected each subject interest to be non-empty.')
+            if not re.match(constants.TAG_REGEX, interest):
+                raise utils.ValidationError(
+                    'Expected each subject interest to consist only of '
+                    'lowercase alphabetic characters and spaces.')
+
+        if len(set(self.subject_interests)) != len(self.subject_interests):
+            raise utils.ValidationError(
+                'Expected each subject interest to be distinct.')
+
+        if not isinstance(self.user_bio, str):
```

Comment:
@AkashPaloju We have a problem -- some user_bio fields on the server are null. This results in the entire server throwing a 500 for some users when this PR is deployed. (In core/storage it looks like required=True is not set for user_bio.)

This validation actually gets enforced on page retrieval because we update the 'user last logged in' value.

For now could you please drop these validations and open a new issue to run audit jobs on the server to identify and fix any issues before introducing the validations? The alternative would be to revert the PR so that we can do the deployment. Thanks.

Reply:
@seanlip , For now I would like to create a new PR which removes this validations and their tests.  I guess, then there will be no errors in test server and you can proceed further for the deployment. Then I will open an issue as you said.

Reply:
OK, sounds good -- thanks @AkashPaloju. Please do this soonish since we need to unblock the server for testing.